internals/core_models.py

```diff
@@ -0,0 +1,237 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file defineds NDB models for the core objects of our system:
+# FeatureEntry, Stage, MilestoneSet, Gate, ApprovalValue.
+
+import logging
+
+from google.cloud import ndb
+
+from framework import ramcache
+from framework import users
+
+
+class FeatureEntry(ndb.Model):  # Copy from Feature
+  """This is the main representation of a feature that we are tracking."""
+
+  # Metadata: Creation and updates.
+  created = ndb.DateTimeProperty(auto_now_add=True)
+  updated = ndb.DateTimeProperty()
+  accurate_as_of = ndb.DateTimeProperty()
+  creator = ndb.StringProperty()
+  updater = ndb.StringProperty()
+
+  # Metadata: Access controls
+  owners = ndb.StringProperty(repeated=True)  # copy from owner
+  editors = ndb.StringProperty(repeated=True)
+  unlisted = ndb.BooleanProperty(default=False)
+  deleted = ndb.BooleanProperty(default=False)
+
+  # Descriptive info.
+  name = ndb.StringProperty(required=True)
+  summary = ndbTtextProperty(required=True)
+  category = ndb.IntegerProperty(required=True)
+  blink_components = ndb.StringProperty(repeated=True)
+  star_count = ndb.IntegerProperty(default=0)
+  search_tags = ndb.StringProperty(repeated=True)
+  feature_notes = ndb.TextProperty()  # copy from comments
+
+  # Metadata: Process information
+  feature_type = ndb.IntegerProperty(default=FEATURE_TYPE_INCUBATE_ID)
+  intent_stage = ndb.IntegerProperty(default=0)
+  bug_url = ndb.StringProperty()
+  launch_bug_url = ndb.StringProperty()
+
+  # Implementation in Chrome
+  impl_status_chrome = ndb.IntegerProperty(required=True)
+  flag_name = ndb.StringProperty()
+  ongoing_constraints = ndb.TextProperty()
+
+  # Gate: Adoption
+  motivation = ndb.TextProperty()
+  devtrial_instructions = ndb.TextProperty()
+  activation_risks = ndb.textProperty()
+  measurement = ndb.TextProperty()
+
+  # Gate: Standardization & Interop
+  initial_public_proposal_url = ndb.StringProperty()
+  explainer_links = ndb.StringProperty(repeated=True)
+  requires_embedder_support = ndb.BooleanProperty(default=False)
+  standard_maturity = ndb.IntegerProperty(required=True, default=UNSET_STD)
+  spec_link = ndb.StringProperty()
+  api_spec = ndb.BooleanProperty(default=False)
+  spec_mentors = ndb.StringProperty(repeated=True)
+  interop_compat_risks = ndb.TextProperty()
+  prefixed = ndb.BooleanProperty()
+  all_platforms = ndb.BooleanProperty()
+  all_platforms_descr = ndb.TextProperty()
+  tag_review = ndb.StringProperty()
+  tag_review_status = ndb.IntegerProperty(default=REVIEW_PENDING)
+  non_oss_deps = ndb.TextProperty()
+  origin_trial_feedback_url = ndb.StringProperty()
```

Comment:
The feedback url might need to live at the stage level, unless all trials associated with a feature assuredly give feedback through the same channel.

Reply:
I am not 100% sure but I think that the origin_trial_feedback_url is supposed to be a document that summarizes the feedback that was given, not a way for users to submit feedback.   If that understanding is correct, then it could be OK to have just one URL for the feature, but I suppose that it would still be OK to have it per-origin-trial.  I'll change it to be in the stage.